File: DrivingRecorder-iOS/DrivingRecorder/Models/EventType.swift
import Foundation

/// 驾驶事件类型
enum EventType: String, Codable, CaseIterable {
    case tripStart = "TRIP_START"
    case tripEnd = "TRIP_END"
    case laneChangeLeft = "LANE_CHANGE_LEFT"
    case laneChangeRight = "LANE_CHANGE_RIGHT"
    case hardBraking = "HARD_BRAKING"
    case rapidAcceleration = "RAPID_ACCELERATION"
    case sharpTurnLeft = "SHARP_TURN_LEFT"
    case sharpTurnRight = "SHARP_TURN_RIGHT"

    var displayName: String {
        switch self {
        case .tripStart: return "行程开始"
        case .tripEnd: return "行程结束"
        case .laneChangeLeft: return "向左变道"
        case .laneChangeRight: return "向右变道"
        case .hardBraking: return "急刹车"
        case .rapidAcceleration: return "急加速"
        case .sharpTurnLeft: return "左急转弯"
        case .sharpTurnRight: return "右急转弯"
        }
    }

    var category: EventCategory {
        switch self {
        case .tripStart, .tripEnd: return .system
        case .laneChangeLeft, .laneChangeRight: return .laneChange
        case .hardBraking, .rapidAcceleration: return .speedChange
        case .sharpTurnLeft, .sharpTurnRight: return .turn
        }
    }
}

enum EventCategory: String {
    case system, laneChange, speedChange, turn
}

File: DrivingRecorder/app/src/main/java/com/drivingrecorder/domain/model/EventType.kt
package com.drivingrecorder.domain.model

/**
 * 驾驶事件类型枚举
 */
enum class EventType(val displayName: String, val category: EventCategory) {
    TRIP_START("行程开始", EventCategory.SYSTEM),
    TRIP_END("行程结束", EventCategory.SYSTEM),

    LANE_CHANGE_LEFT("向左变道", EventCategory.LANE_CHANGE),
    LANE_CHANGE_RIGHT("向右变道", EventCategory.LANE_CHANGE),

    HARD_BRAKING("急刹车", EventCategory.SPEED_CHANGE),
    RAPID_ACCELERATION("急加速", EventCategory.SPEED_CHANGE),

    SHARP_TURN_LEFT("左急转弯", EventCategory.TURN),
    SHARP_TURN_RIGHT("右急转弯", EventCategory.TURN);

    companion object {
        fun fromName(name: String): EventType =
            entries.firstOrNull { it.name == name } ?: TRIP_START
    }
}

enum class EventCategory {
    SYSTEM,
    LANE_CHANGE,
    SPEED_CHANGE,
    TURN
}
